js/InputReader.js

```diff
@@ -0,0 +1,64 @@
+export class InputReader {
+    #input
+
+    constructor(input) {
+        if (typeof(input) !== "string") {
+            throw new TypeError("Expected input to be a string.")
+        }
+
+        this.#input = input
+    }
+
+    /**
+     * Iterates through the input string and yields all contained tokens.
+     *
+     * A token is either a Number representing an operand,
+     * or a string representing an operator or a bracket.
+     * E.g. the input "(3+45)" yields: "(", 3, "+", 45, ")"
+     */
+    *iterateTokens() {
+        for (let i = 0; i < this.#input.length; i++) {
+            if (InputReader.#isNumber(this.#input[i])) {
+                let number = InputReader.#completeNumberParser(this.#input, i)
+                i += InputReader.#lengthOfNumber(number) - 1
+                yield parseInt(number)
+            } else if (this.#input[i] === " ") {
+                continue
+            } else {
+                yield this.#input[i]
+            }
+        }
+    }
+
+    static #isNumber(char) {
+        const numbers = [ "0", "1", "2", "3", "4", "5", "6", "7", "8", "9" ]
+        for (const i of numbers) {
+            if (char === i) {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    static #completeNumberParser(input, i) {
+        let result = 0
+        while (InputReader.#isNumber(input[i])) {
+            result *= 10
+            result += parseInt(input[i])
+            i += 1
+        }
+
+        return result
+    }
+
+    static #lengthOfNumber(number) {
```

Comment:
This can be a one-liner if you use [Number.toString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString). 

Also, the name could be shortened to `#lengthOf(number)`.